src/formatDate.js

```diff
@@ -50,7 +53,65 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const prefixYear = (yearBefore, newFormat) => {
+    const yearUpdateFormat
+      = newFormat.filter(partDate => partDate.includes('YY'))[0];
+
+    const previousCentury = '19';
+    const currentCentury = '20';
+
+    if (yearBefore.toString().length > yearUpdateFormat.length) {
+      return yearBefore.slice(-2);
+    }
+
+    if (yearBefore.length === yearUpdateFormat.length) {
+      return yearBefore;
+    }
+
+    if (yearUpdateFormat.length === 4 && yearBefore >= 30) {
+      return previousCentury + yearBefore;
+    }
+
+    if (yearUpdateFormat.length === 4 && yearBefore < 30) {
+      return currentCentury + yearBefore;
+    }
+
+    return yearBefore;
+  };
+
+  let newFormatDate = '';
+
+  for (let i = 0; i < toFormat.length; i++) {
+    const dateArr = date.split(fromFormat[3]);
+    const indexYY = fromFormat.indexOf('YY') !== -1
+      ? fromFormat.indexOf('YY')
+      : fromFormat.indexOf('YYYY');
+
+    const indexMM = fromFormat.indexOf('MM');
+    const indexDD = fromFormat.indexOf('DD');
+
+    const day = dateArr[indexDD];
+    const month = dateArr[indexMM];
+    const year = dateArr[indexYY];
+
+    const separator = toFormat[toFormat.length - 1];
+
+    if (toFormat[i] === 'DD') {
+      newFormatDate += day + separator;
+    }
+
+    if (toFormat[i] === 'MM') {
+      newFormatDate += month + separator;
+    }
+
+    if (toFormat[i] === 'YY' || toFormat[i] === 'YYYY') {
+      const yearUpdate = prefixYear(year, toFormat);
+
+      newFormatDate += yearUpdate + separator;
+    }
+  }
+
+  return newFormatDate.slice(0, newFormatDate.length - 1);
```

Comment:
I think it will be better to set newFormatDate to array and just add values as string by index and then return this array.join(separator). You will get rid of repeating separator and your return statement will be easier to understand.

Reply:
it seems obvious to me that slice newFormatDate. length - 1 I just don't include the last separator.

So your suggestion will make the code shorter, but I can't see the difference for reading))